tests/core/test_linalg.py

```diff
@@ -125,6 +125,19 @@ def test_inv(app_inst: ArrayApplication):
         assert np.allclose(np.linalg.cholesky(mat.get()), L, rtol=1e-4, atol=1e-4)
 
 
+def test_inv_uppertri(app_inst: ArrayApplication):
+    block_shape, shape = (3, 3), (20, 20)
```

Comment:
i was wondering if it would be a good idea to test with some random ```block_shapes``` and ```shapes``` and then loop over those using ```itertools.product```

Reply:
For now, just add a #TODO here for this